src/Input/Password.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace PhpSchool\CliMenu\Input;
+
+use PhpSchool\CliMenu\MenuStyle;
+
+/**
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class Password implements Input
+{
+    /**
+     * @var InputIO
+     */
+    private $inputIO;
+
+    /**
+     * @var string
+     */
+    private $promptText = 'Enter password:';
+
+    /**
+     * @var string
+     */
+    private $validationFailedText = 'Invalid password, try again';
+
+    /**
+     * @var string
+     */
+    private $placeholderText = '';
+
+    /**
+     * @var null|callable
+     */
+    private $validator;
+
+    /**
+     * @var MenuStyle
+     */
+    private $style;
+
+    public function __construct(InputIO $inputIO, MenuStyle $style)
+    {
+        $this->inputIO = $inputIO;
+        $this->style = $style;
+    }
+
+    public function setPromptText(string $promptText) : Input
+    {
+        $this->promptText = $promptText;
+
+        return $this;
+    }
+
+    public function getPromptText() : string
+    {
+        return $this->promptText;
+    }
+
+    public function setValidationFailedText(string $validationFailedText) : Input
+    {
+        $this->validationFailedText = $validationFailedText;
+
+        return $this;
+    }
+
+    public function getValidationFailedText() : string
+    {
+        return $this->validationFailedText;
+    }
+
+    public function setPlaceholderText(string $placeholderText) : Input
+    {
+        $this->placeholderText = $placeholderText;
+
+        return $this;
+    }
+
+    public function getPlaceholderText() : string
+    {
+        return $this->placeholderText;
+    }
+
+    public function setValidator(callable $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    public function ask() : InputResult
+    {
+        return $this->inputIO->collect($this);
+    }
+
+    public function validate(string $input) : bool
+    {
+        if ($this->validator) {
+            $validator = $this->validator;
+            return $validator($input);
```

Comment:
1 thing I noticed here was that a developer could potentially have quite a complex validator added to an input with 1 or more "rules"... however there's no way to feedback from that validator the error that occurred, in this case they'd need to create a new input type 

What do you think, worth refactoring a little for that ?

Reply:
Yeah I thought about this but was being lazy. Probably allow to throw an exception and use the exception message, bit like how symfony does.